linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -278,14 +278,15 @@ where
         if query.request_committees {
             info.requested_committees = Some(chain.execution_state.system.committees.get().clone());
         }
-        match query.request_owner_balance {
-            owner @ AccountOwner::Application(_) | owner @ AccountOwner::User(_) => {
-                info.requested_owner_balance =
-                    chain.execution_state.system.balances.get(&owner).await?;
-            }
-            AccountOwner::Chain => {
-                info.requested_owner_balance = Some(*chain.execution_state.system.balance.get());
-            }
+        if query.request_owner_balance == AccountOwner::chain() {
+            info.requested_owner_balance = Some(*chain.execution_state.system.balance.get());
```

Comment:
We could make this another entry in `balances` now.

Reply:
Sadly, there is an issue with lifetimes in `ResourceController::with_state_and_grant`
https://github.com/linera-io/linera-protocol/blob/main/linera-execution/src/resources.rs#L524